java/client/src/main/java/compatibility/clients/jedis/Jedis.java

```
@@ -387,4 +408,1394 @@ public Set<String> keys(String pattern) {
             throw new JedisException("KEYS operation failed", e);
         }
     }
+
+    // ===== STRING COMMANDS =====
+
+    /**
+     * Set multiple key-value pairs.
+     *
+     * @param keysvalues alternating keys and values
+     * @return "OK"
+     */
+    public String mset(String... keysvalues) {
+        checkNotClosed();
+        try {
+            Map<String, String> keyValueMap = new HashMap<>();
+            for (int i = 0; i < keysvalues.length; i += 2) {
```

If keysvalues has an odd number of entries, this will silently truncate the last element. Does Jedis do this? Should we throw an IllegalArgumentException instead?

---

Updated. Added a check with an exception above the map declaration.